ext/introduction.asciidoc

```diff
@@ -121,7 +121,18 @@ The initial state of the compiler is as if the directive
 ----
 
 was issued, telling the compiler that all error and warning reporting must
-be done according to this specification, ignoring any extensions.
+be done according to the core specification, ignoring any extensions.
+
+Extension pragma directives are optional and they are only required for
+extensions that change the compiler behavior i.e. the source code is interpreted
+differently when the pragma directive is used. Each extension should specify
+whether it adds the directive or not.  There is no default behavior for
+the extension pragma directive that would apply to extensions universally e.g.
+it does not define or undefine an associated predefined extension
+macro; it does not add or remove function or type declarations. Implementations
+are not required to accept extension pragmas if their behavior is not specified
+explicitly. The use of extension pragma with unknown behavior does not cause
+the compilation to fail but might result in a diagnostic.
```

Comment:
This description is good but it is subtly different than the description in the OpenCL C spec.  Would it be better to keep them more in sync?  Or, should the OpenCL C spec description be shortened further and refer to this description in the extension spec for more detail?

I especially like the way that this description clarifies that unknown extension pragmas will not cause the compilation to fail but might result in a diagnostic.

Reply:
One other observation: the table above (line 91) says:

> Report an error on the #pragma OPENCL EXTENSION if the extension_name is not supported, or if all is specified.

Is this still correct and should an error be reported?  Or, is an unsupported extension considered to be unknown, meaning it may result in a diagnostic but will not fail compilation?

Reply:
The way I understood this table is that it applies to extensions where the pragma `enable`/`disable` is needed i.e. actually doing something. If this is the case then if the extension is not supported but the pragma is attempting to enable it the error is sensible. However, I am not sure whether this is achievable to implement because compilers can't possibly know all extensions that exist and we don't want them to know all of them. So if there is let's say an extension that is implemented by only one vendor and not added into any other compiler then the other compilers wouldn't know about such extension and therefore wouldn't report that it is not supported. They would only be able to report that the extension is unknown. It seems that the definition of unknown and unsupported extension here is somewhat vague.

Perhaps, it is reasonable to change this to say that a diagnostic will occur w/o detailing which exactly diagnostic?